storage/fakestorage.py

```diff
@@ -45,6 +45,9 @@ def put_content(self, path, content):
         self._fake_storage_map[path].write(content)
 
     def stream_read(self, path):
+        if not self.exists(path):
```

Comment:
A call to `stream_read` would result in the `path` getting added to `self._fake_storage_map`. I'm not sure why it works like this, but the behaviour is odd and it made one of my tests pass when it should have failed (I was expecting the blob to not exist in storage until it was actually added to it).

So this check was added to make the fake storage both more predictable and more real-life-like.